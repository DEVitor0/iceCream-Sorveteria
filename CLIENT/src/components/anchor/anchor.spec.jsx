import React from 'react';
import { render, screen } from '@testing-library/react';
import Anchor from './index';

test('renders anchor with correct text and href', () => {
  const { asFragment } = render(
    <Anchor href="https://example.com" text="Go to Example" />,
  );

  const anchorElement = screen.getByText(/Go to Example/i);
  expect(anchorElement).toBeInTheDocument();

  const linkElement = screen.getByRole('link');
  expect(linkElement).toHaveAttribute('href', 'https://example.com');

  expect(asFragment()).toMatchSnapshot();
});

test('anchor has correct class for styling', () => {
  const { asFragment } = render(
    <Anchor href="https://example.com" text="Go to Example" />,
  );

  const linkElement = screen.getByRole('link');
  expect(linkElement).toHaveClass('anchor-style');

  expect(asFragment()).toMatchSnapshot();
});
