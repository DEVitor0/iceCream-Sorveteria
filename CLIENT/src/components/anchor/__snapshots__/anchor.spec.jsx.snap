// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anchor has correct class for styling 1`] = `
<DocumentFragment>
  <a
    class="anchor-style"
    href="https://example.com"
  >
    Go to Example
  </a>
</DocumentFragment>
`;

exports[`renders anchor with correct text and href 1`] = `
<DocumentFragment>
  <a
    class="anchor-style"
    href="https://example.com"
  >
    Go to Example
  </a>
</DocumentFragment>
`;
